pkg/condition/condition.go

```diff
@@ -21,3 +21,14 @@ const (
 	InstallationFailed        Reason = "InstallationFailed"
 	Pending                   Reason = "Pending"
 )
+
+func (r Reason) Recoverable() bool {
+	switch r {
+	case InstallationFailed:
+		return false
+	case UnsupportedFormat:
+		return false
+	default:
+		return true
+	}
```

Comment:
I don't think defaulting to recoverable is a good idea. The goal of this feature would be having the ability to consider some errors as recoverable, which is different from the current behavior where all errors are considered non-recoverable.